addons/cert.py

```diff
@@ -247,7 +250,30 @@ def str05(data):
             if parent.isAssignmentOp and parentOp1.valueType:
                 if (parentOp1.valueType.type in ('char', 'wchar_t')) and parentOp1.valueType.pointer and not parentOp1.valueType.constness:
                     reportError(parentOp1, 'style', 'Use pointers to const when referring to string literals', 'STR05-C')
-                
+
+# STR07-C
+# Use the bounds-checking interfaces for string manipulation
+def str07(data):
+    for token in data.tokenlist:
+        if token.str in('strcpy', 'strcat'):
+            args = cppcheckdata.getArguments(token)
+            if len(args)!=2:
+                continue
+            if args[1].variable is None:
+                continue
+
+            variableUninit=False
+            for value in args[1].values:
+                if value.uninit is None:
+                    continue
+                variableUninit = value.uninit
+                break
+
+            if args[1].isName and args[1].variable.isPointer and variableUninit and token.str=='strcpy':
```

Comment:
I would prefer common conditions instead of repeating the same code for 'strcpy' and 'strcat'.